deployments.go

```diff
@@ -92,9 +92,27 @@ func (s *deployerService) Deploy(ctx context.Context, opts DeploymentsCreateOpts
 		Description: desc,
 	})
 
-	if err := json.NewEncoder(opts.Output).Encode(&jsonmessage.JSONMessage{
-		Status: fmt.Sprintf("Status: Created new release v%d for %s", r.Version, r.App.Name),
-	}); err != nil {
+	if err != nil {
+		return nil, err
+	}
+
+	return r, nil
+}
+
+// Deploy is a thin wrapper around doDeploy to handle errors & output more cleanly
+func (s *deployerService) Deploy(ctx context.Context, opts DeploymentsCreateOpts) (*Release, error) {
+	r, err := s.doDeploy(ctx, opts)
+	var msg jsonmessage.JSONMessage
+	if err != nil {
+		msg = newJSONMessageError(err)
+		if err = json.NewEncoder(opts.Output).Encode(&msg); err != nil {
```

Comment:
This would result in the error being encoded into the stream twice, no? Once here, then it falls down to the one below. I think you just want this:

```go
var (
        msg jsonmessage.JSONMessage
        enc = json.NewEncoder(opts.Output)
)

r, err := s.doDeploy(ctx, opts)
if err != nil {
        msg = newJSONMessageError(err)
} else {
        msg = jsonmessage.JSONMessage{Status: fmt.Sprintf("Status: Created new release v%d for %s", r.Ver
}

if err := enc.Encode(&msg); err != nil {
        return r, err
}

return r, err
```